### PR TITLE
Add Help command functionality

### LINKED ISSUE #105: Implement help command for all commands

### PR DESCRIPTION
**Pull request by ElginL**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -89,17 +89,19 @@ interactions recorded faster and simpler than traditional GUI applications.
 
 ### Adding a company: `add`
 
-Adds a point-of-contact to the company.
+Adds an empty company to the list without any POC (Point-Of-Contact) and transactions.
 
-Format: `add coy/COMPANY n/NAME hp/NUMBER e/EMAIL loc/ADDRESS`
+Format: `add n/NAME loc/ADDRESS`
```
Comment (notle1706) [inline docs/UserGuide.md]: Should the prefix for address be `a/` instead of `loc/`?
This should be easier to implement since AB3 uses `a/`.

**Review by notle1706 — APPROVED** [review]
Everything looks good. Perhaps you could consider the suggestion to change the prefix of address to `a/`?

File: docs/UserGuide.md (hunk 1 above)
Comment (ElginL) [reply docs/UserGuide.md]: Thanks for the feedback, I've just updated it to `a/`

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.commands;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Helps the user to understand what other commands do, and how to use it.
```
Comment (jicsontoh) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:7]: Clear and concise description of class

File: src/main/java/seedu/address/logic/parser/HelpCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+public class HelpCommandParser implements Parser<HelpCommand> {
+
+    public static final String INVALID_ARGUMENT_MESSAGE = "Command you are trying to get help for is invalid!\n"
+            + "Use 'help' to view valid commands";
+
+    @Override
+    public HelpCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim().toLowerCase();
+
+        switch (trimmedArgs) {
```
Comment (jicsontoh) [inline src/main/java/seedu/address/logic/parser/HelpCommandParser.java:25]: Good indentation for switch statements

**Review by jicsontoh — APPROVED** [review]
LGTM

**Pull request merged**